stdlib/task/src/main/java/org/ballerinalang/stdlib/task/utils/Utils.java

```diff
@@ -0,0 +1,135 @@
+import static org.ballerinalang.stdlib.task.utils.TaskConstants.TASK_ERROR_MESSAGE;
+import static org.ballerinalang.stdlib.task.utils.TaskConstants.TASK_ERROR_RECORD;
+import static org.quartz.CronExpression.isValidExpression;
+
+/**
+ * Utility functions used in ballerina task module.
+ */
+public class Utils {
+
+    public static BError createError(Context context, String message) {
+        BMap<String, BValue> taskErrorRecord = createTaskErrorRecord(context);
+        taskErrorRecord.put(TASK_ERROR_MESSAGE, new BString(message));
+        return BLangVMErrors.createError(context, true, BTypes.typeError, TASK_ERROR_CODE, taskErrorRecord);
+    }
+
+    private static BMap<String, BValue> createTaskErrorRecord(Context context) {
+        return BLangConnectorSPIUtil.createBStruct(context, PACKAGE_STRUCK_NAME, TASK_ERROR_RECORD);
+    }
+
+    public static String getCronExpressionFromAppointmentRecord(BValue record)
+            throws SchedulingException {
+
+        String cronExpression;
+        if (record instanceof BMap && STRUCT_APPOINTMENT_DATA.equals(record.getType().getName())) {
+            cronExpression = buildCronExpression((BMap) record);
+            if (!isValidExpression(cronExpression)) {
+                throw new SchedulingException("AppointmentData \"" + record.stringValue() + "\" is invalid.");
+            }
+        } else {
+            cronExpression = record.stringValue();
+            if (!isValidExpression(cronExpression)) {
+                throw new SchedulingException("Cron Expression \"" + cronExpression + "\" is invalid.");
+            }
+        }
+        return cronExpression;
+    }
+
+    // Following code is reported as duplicates since all the lines doing same function call.
+    private static String buildCronExpression(BMap<String, BValue> record) {
+        String cronExpression = getStringFieldValue(record, FIELD_SECONDS) + " " +
+                getStringFieldValue(record, FIELD_MINUTES) + " " +
+                getStringFieldValue(record, FIELD_HOURS) + " " +
+                getStringFieldValue(record, FIELD_DAYS_OF_MONTH) + " " +
+                getStringFieldValue(record, FIELD_MONTHS) + " " +
+                getStringFieldValue(record, FIELD_DAYS_OF_WEEK) + " " +
+                getStringFieldValue(record, FIELD_YEAR);
+        return cronExpression.trim();
+    }
+
+    private static String getStringFieldValue(BMap<String, BValue> struct, String fieldName) {
+        if (FIELD_DAYS_OF_MONTH.equals(fieldName) && Objects.isNull(struct.get(FIELD_DAYS_OF_MONTH))) {
+            return "?";
+        } else if (Objects.nonNull(struct.get(fieldName))) {
+            return struct.get(fieldName).stringValue();
+        } else {
+            return "*";
+        }
+    }
+
+    /*
+     * TODO: Runtime validation is done as compiler plugin does not work right now.
+     *       When compiler plugins can be run for the resources without parameters, this will be redundant.
+     */
```

Comment:
Do we need this TODO? If yes, pls create a github issue to track this and add the details here

Reply:
Done